File: 0342-power-of-four/0342-power-of-four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n == 1: return True
        if n <= 0: return False
        n, x = abs(n), 1
        while 4 ** x < n: x += 1
        return True if 4 ** x == n else False

File: 0342-power-of-four/0342-power-of-four.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n <= 0)
            return false;
        if (n > 1 && n % 4 == 0) {
            return isPowerOfFour(n / 4);
        } else if (n > 1 && (n % 4) != 0) {
            return false;
        }
        return true;
    }
};